# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/difftastic/guess.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- difftastic-0.41.0/src/parse/guess_language.rs.vanilla	2023-01-11 18:24:50.793566876 +0100
+++ difftastic-0.41.0/src/parse/guess_language.rs	2023-01-11 18:25:36.193566724 +0100
@@ -129,7 +129,7 @@
     (
         Bash,
         &[
-            "sh", "bash", "bats", "cgi", "command", "env", "fcgi", "ksh", "sh.in", "tmux", "tool",
+            "sh", "bash", "bats", "cgi", "command", "env", "fcgi", "ksh", "in", "tmux", "tool",
             "zsh",
         ],
     ),
@@ -145,7 +145,7 @@
     // Treat .h as C++ rather than C. This is an arbitrary choice, but
     // C++ is more widely used than C according to
     // https://madnight.github.io/githut/
-    (CPlusPlus, &["cc", "cpp", "h", "hh", "hpp", "cxx"]),
+    (CPlusPlus, &["cc", "cpp", "h", "hh", "hpp", "cxx", "tcc"]),
     (CSharp, &["cs"]),
     (Css, &["css"]),
     (Dart, &["dart"]),
@@ -191,7 +191,7 @@
     (Jsx, &["jsx"]),
     (Julia, &["jl"]),
     (Kotlin, &["kt", "ktm", "kts"]),
-    (Lua, &["lua"]),
+    (Lua, &["lua", "els"]),
     (Make, &["mak", "d", "make", "makefile", "mk", "mkfile"]),
     (Nix, &["nix"]),
     (OCaml, &["ml"]),
